llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -789,6 +791,23 @@ def UMAX16x2 : I16x2<"max.u", umax>;
 def SMIN16x2 : I16x2<"min.s", smin>;
 def UMIN16x2 : I16x2<"min.u", umin>;
 
+let Predicates = [hasPTX<80>, hasSM<90>, hasOptEnabled] in {
```

Comment:
Is there a particular reason to keep lowering to max.relu behind optimization level?
We virtually never do that for other patterns we match to an instruction.

The way I see it, the job of the lowering is to always convert the DAG to the best instructions available to represent it. User-controlled optimizations generally belong to the higher levels.  I would prefer not to add another moving piece into the lowering process.

AFAICT TM.getOptLevel() is mostly used to configure which passes get enabled/disabled, and NVPTX is the only back-end which uses it in the back-end tablegen file.